File: python/28. Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        s = haystack
        p = needle
        i = 0
        ls, lp = len(haystack), len(needle)
        while i <= ls - lp:
            j = 0
            while j < lp:
                if s[i+j] != p[j]:
                    break
                j += 1
            if j == lp:
                return i
            i += 1
        return -1
        

File: go/src/leetcode/28. Implement strStr()/28. Implement strStr().go
package main

func strStr(haystack string, needle string) int {
	if needle == haystack {
		return 0
	}
	m, n := len(haystack), len(needle)
	if n > m {
		return -1
	}
	for i := 0; i <= m-n; i++ {
		if haystack[i:i+n] == needle {
			return i
		}
	}
	return -1
}
